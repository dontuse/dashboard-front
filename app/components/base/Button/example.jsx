/* eslint no-console: 0 */
/* eslint no-alert: 0 */
import React from 'react';
import Button from './Button';

export default class ButtonExample extends React.Component {

  render() {
    return (
      <div>
        <h3>Button:</h3>
        <Button onClick={() => { alert('button 1 clicked'); }}>ДОБАВИТЬ ГРУППУ</Button> {' '}
        <Button>Отмена</Button> <Button mix='is-good'>Сохранить и продолжить</Button> {' '}
        <Button disabled onClick={() => { alert('button 2 clicked'); }} mix='test-mix'>
          disabled
        </Button> {' '}
        <Button
          onClick={() => {}}
          mix='is-big-size'
        >
          Большая кнопка
        </Button> {' '}
        <Button
          onClick={() => {}}
          mix='is-cancel is-big-size'
        >
          Большая кнопка
        </Button>
      </div>
    );
  }
}
