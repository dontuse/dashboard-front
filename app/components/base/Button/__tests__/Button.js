import React from 'react';
import {shallow, mount, render} from 'enzyme';
import Button from '../Button';

describe('Button', () => {
  it('should render a Button', () => {
    const wrapper = shallow(
      <Button>Some text</Button>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('Button with css class mix', () => {
    const wrapper = shallow(
      <Button mix="some-mix">Some text</Button>
    );
    expect(wrapper.find('.some-mix')).toHaveLength(1);
  });

  it('Button should be clickable', () => {
    const onClick = jest.fn();
    const wrapper = shallow(<Button onClick={onClick} />);

    wrapper.find('button').simulate('click', {preventDefault: jest.fn()});

    expect(onClick).toBeCalled();
});
});
