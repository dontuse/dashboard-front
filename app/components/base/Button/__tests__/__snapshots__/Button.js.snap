// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render a Button 1`] = `
<button
  className={[Function]}
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Some text
</button>
`;
